### ISSUE TITLE
MicStream error on MicStartRecording

### ISSUE THREAD

**Issue by onsterion** [Microsoft/MixedRealityToolkit-Unity#993]
Hi All,

Im testing the MicStreamDemo , works perfect in Windows PC but in Hololens when i try to use: 
"CheckForErrorOnCall(MicStream.MicStartRecording(SaveFileName, false));".

I have this error:  
"Exception thrown: 'System.Runtime.InteropServices.SEHException' in Assembly-CSharp.dll
An exception of type 'System.Runtime.InteropServices.SEHException' occurred in Assembly-CSharp.dll but was not handled in user code
External component has thrown an exception.  "

And in the "outputPath = MicStream.MicStopRecording();" I receive: "YouArentRecordingRightNow"

Unity Version: 2017.1.1p1 x64
Hololens System: Full updated 09/19/2017

Any suggestion?

Regards.-

**Comment by StephenHodgson** [comment]
Just FYI, that dll hasn't been well maintained and I'm planning on removing it from the core MRTK.
I don't believe it's passing WACK compliance at the moment.

**Comment by Ziugy** [comment]
@StephenHodgson do you have, or know of, a replacement for the microphone input if you remove it?

**Comment by StephenHodgson** [comment]
No. But a lot of that stuff is written into unity already.

**Comment by onsterion** [comment]
Thanks StephenHodgson for the quick response.

**Comment by joekellyms** [comment]
Unity's Microphone class is a lot more limited than the plugin. Getting it to pass WACK would be much preferred to removing it.

**Comment by StephenHodgson** [comment]
@joekellyms, yeah I agree but getting PRs addressed and updated in the main toolkit seems impossible.

**Comment by tberthin** [comment]
Hi @StephenHodgson 
I'm in the same situation as @onsterion, same versions, same error (I tried multiple versions actually, without success) and I don't find any other way to record microphone input from Hololens, would you have a solution or a hint about how to do this ?
Should I try from MicStartStream (which works apparently), from source code or am I missing something ?
Thank you.

**Comment by StephenHodgson** [comment]
Well, I know this doesn't help, but I've got plans on removing this from the core toolkit cause it isn't passing WACK.

**Comment by tberthin** [comment]
@StephenHodgson Could it be possible to access data from MicStartStream (since variable "keepData" is described as "If true, all data will stay in the queue,...") without modifying MicStreamSelector.cpp (because I don't feel being able to) in order to record that data ? 
Sorry for insisting I'm a bit confused failing to record that mic.

**Comment by davidkline-ms** [comment]
I experience this when i do NOT have the Music Library capability set. If I enable the bit, the application works as expected.  The simple fix is to enable Music Library capability.

I am also filing an issue on the plugin that the native code side does not appear to be doing the right thing.

**Comment by keveleigh** [comment]
Fixed via #1670 and Microsoft/MixedRealityToolkit#131

**Comment by StephenHodgson** [comment]
We will close these once they make it into the release master branch.
